proxy/coupon.model.js

```diff
@@ -33,4 +34,45 @@ exports.createCouponWithDefaultRulesForSpecifiedUser = function (username, coupo
   var newCoupon = {username : username, couponID : couponId,
     couponRule : defaultRules.couponRule, rebateRule : defaultRules.rebateRule};
   return new Coupon(newCoupon).save();
+};
+
+exports.isBelongToUsers = function (couponId, username) {
+  return Coupon.find({ couponID : couponId} ).then (function (coupons){
+    return coupons[0].username === username;
+  }, function (err) {
+    return err;
+  });
+}
+
+exports.getDiscountedValue = function (couponId, orderValue) {
+  return Coupon.find({ couponID : couponId} ).then (function (coupons){
+    var couponObject = coupons[0];
+    console.log(couponObject + '---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------')
+
+    var ruleType = couponObject.couponRule.type;
+    var ruleValue = couponObject.couponRule.value;
+    var dicountedValue = 0 ;
+    console.log(ruleType + ruleValue + orderValue + dicountedValue + '---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------')
+ 
+    if (ruleType === 'PERCENTAGE')
+    {
+      dicountedValue = orderValue * (100 - ruleValue) / 100;
+    }
+    else if (ruleType === 'CASH') 
+    {
+      dicountedValue = orderValue - ruleValue;
+    }
+    else
+    {
+      dicountedValue = orderValue;
+    }
+    dicountedValue = parseInt(dicountedValue);
+    console.log(dicountedValue + '---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------')
+
+    return  { 'couponID': couponObject.couponID, 'dicountedValue' : dicountedValue };
+
+  }).then(function (discountedOrder) {
+    console.log(discountedOrder + '---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------')
+    return discountedOrder;
+  });
 };
```

Comment:
I'd think this is part of the business logic, maybe it's better move to order to the router, but it's up to you.